platform/features/conductor/core/src/ui/TimeConductorController.js

```diff
@@ -22,218 +22,259 @@
          * @private
+         * @param key
+         * @returns {TimeSystem}
          */
-        TimeConductorController.prototype.initializeScope = function () {
-            //Set time Conductor bounds in the form
-            this.$scope.boundsModel = this.conductor.bounds();
-
-            //If conductor has a time system selected already, populate the
-            //form from it
-            this.$scope.timeSystemModel = {};
-
-            //Represents the various modes, and the currently selected mode
-            //in the view
-            this.$scope.modeModel = {
-                options: this.conductorViewService.availableModes()
-            };
-
-            // Watch scope for selection of mode or time system by user
-            this.$scope.$watch('modeModel.selectedKey', this.setMode);
-
-            this.conductorViewService.on('pan', this.onPan);
-            this.conductorViewService.on('pan-stop', this.onPanStop);
-
-            this.$scope.$on('$destroy', this.destroy);
-        };
-
-        TimeConductorController.prototype.setStateFromSearchParams = function (searchParams) {
-            //Set mode from url if changed
-            if (searchParams[SEARCH.MODE] === undefined ||
-                searchParams[SEARCH.MODE] !== this.$scope.modeModel.selectedKey) {
-                this.setMode(searchParams[SEARCH.MODE] || this.DEFAULT_MODE);
-            }
-
-            if (searchParams[SEARCH.TIME_SYSTEM] &&
-                searchParams[SEARCH.TIME_SYSTEM] !== this.conductor.timeSystem().metadata.key) {
-                //Will select the specified time system on the conductor
-                this.selectTimeSystemByKey(searchParams[SEARCH.TIME_SYSTEM]);
-            }
-
-            var validDeltas = searchParams[SEARCH.MODE] !== 'fixed' &&
-                searchParams[SEARCH.START_DELTA] &&
-                searchParams[SEARCH.END_DELTA] &&
-                !isNaN(searchParams[SEARCH.START_DELTA]) &&
-                !isNaN(searchParams[SEARCH.END_DELTA]);
-
-            if (validDeltas) {
-                //Sets deltas from some form model
-                this.setDeltas({
-                    startDelta: parseInt(searchParams[SEARCH.START_DELTA]),
-                    endDelta: parseInt(searchParams[SEARCH.END_DELTA])
-                });
-            }
-
-            var validBounds = searchParams[SEARCH.MODE] === 'fixed' &&
-                searchParams[SEARCH.START_BOUND] &&
-                searchParams[SEARCH.END_BOUND] &&
-                !isNaN(searchParams[SEARCH.START_BOUND]) &&
-                !isNaN(searchParams[SEARCH.END_BOUND]);
-
-            if (validBounds) {
-                this.conductor.bounds({
-                    start: parseInt(searchParams[SEARCH.START_BOUND]),
-                    end: parseInt(searchParams[SEARCH.END_BOUND])
-                });
-            }
+        TimeConductorController.prototype.getTimeSystem = function (key) {
```

Comment:
This is a minor nit, this method could be removed; in the `optionsFromConfig` you're building an associative array of clocks; so why not enumerate clocks there instead of doing multiple traversals.